crates/bevy_input/src/touch.rs

```diff
@@ -321,24 +327,27 @@ mod test {
         touches.update();
         touches.process_touch_event(&cancel_touch_event);
 
-        assert!(touches.just_cancelled.get(&cancel_touch_event.id).is_some());
-        assert!(touches.pressed.get(&cancel_touch_event.id).is_none());
+        assert!(touches.just_cancelled.get(&touch_event.id).is_some());
+        assert!(touches.pressed.get(&touch_event.id).is_none());
 
         // Test ending an event
 
         let end_touch_event = TouchInput {
             phase: TouchPhase::Ended,
-            position: Vec2::new(4.0, 4.0),
+            position: Vec2::new(5.0, 5.0),
             force: None,
-            id: 4,
+            id: touch_event.id,
         };
 
         touches.update();
         touches.process_touch_event(&touch_event);
+        touches.process_touch_event(&moved_touch_event);
         touches.process_touch_event(&end_touch_event);
 
         assert!(touches.just_released.get(&touch_event.id).is_some());
         assert!(touches.pressed.get(&touch_event.id).is_none());
+        let touch = touches.just_released.get(&touch_event.id).unwrap();
+        assert!(touch.previous_position != touch.position);
```

Comment:
It would be good to add an explanation for this assertion.

Reply:
updated